quiz/quizfinal.js

```diff
@@ -77,132 +70,207 @@ function getRandomQuestions(quizQuestionsArr) {
     const randomQuestionIndex = Math.floor(
       Math.random() * quizQuestionsArr.length
     );
     randomQuiz.push(quizQuestionsArr[randomQuestionIndex]);
     quizQuestionsArr.splice(randomQuestionIndex, 1);
-    // console.log("randomQuiz", randomQuiz);
     quantity++;
-
-    // }
   }
   randomQuiz.sort((a, b) => a.level - b.level);
-  // console.log("randomQuiz", randomQuiz);
 
   return randomQuiz;
 }
 
-// console.log("randomQuiz", randomQuiz);
-
-console.log(getRandomQuestions(quizQuestions.math));
-
-const nextBtn = document.querySelector("#next-btn");
-
 let formElem = document.querySelector("form");
 formElem.addEventListener("submit", (event) => {
   event.preventDefault();
 });
 
-function renderCurrentQuestionToForm(quiz, currentIndex, formElem) {
-  console.log("formElem", formElem);
-  formElem.innerHTML = "";
-  const questionBox = document.createElement("div");
-  questionBox.classList.add("question-box");
-  questionBox.id = "question-box";
-  const questionCountdown = document.createElement("span");
-  // console.log("quiz.length", quiz.length);
-  questionCountdown.innerHTML = `Question ${currentIndex + 1} of ${
-    quiz.length
-  }`;
-  questionBox.append(questionCountdown);
-  const questionElem = document.createElement("p");
-  questionElem.classList.add("question");
-  questionElem.id = "question";
+function renderAnswerOptions(quiz, currentIndex, isResult) {
   // console.log("currentIndex", currentIndex);
-  // console.log("randomQuiz", quiz);
-  // transfer to other function
-  questionElem.innerHTML = quiz[currentIndex]["question"];
-  questionBox.append(questionElem);
+  // console.log("randomQuiz", randomQuiz);
 
-  // answers/ write new function
   const answersArray = quiz[currentIndex]["answersArray"];
-  console.log("answersArray", answersArray);
   const answersList = document.createElement("ul");
   answersList.classList.add("answers");
 
-  for (let arrAnswer in answersArray) {
-    const answerValue = answersArray[arrAnswer]["answer"];
-
+  for (let answerObject of answersArray) {
+    const answerValue = answerObject.answer;
     const answerLine = document.createElement("li");
     answerLine.classList.add("answer");
     answerLine.id = "answer";
     const radioButton = document.createElement("input");
     radioButton.setAttribute("type", "radio");
-    radioButton.name = "radio";
+    radioButton.name = "answer";
+    radioButton.id = "answer";
     radioButton.value = answerValue;
 
     const label = document.createElement("label");
-    label.innerHTML = answerValue;
-    answerLine.append(radioButton);
+    label.append(radioButton);
     answerLine.append(label);
+    answerLine.append(answerValue);
+
+    const currentQuestion = quiz[currentIndex];
+    if (isResult) {
+      // can I move this?
+      if (answerValue === currentQuestion.chosenAnswer) {
+        radioButton.checked = true;
+      } else {
+        radioButton.disabled = true;
```

Comment:
do you want to disable only non chosen answers?

Reply:
yes, bcs we decided to show checked the chosen answer

Reply:
can't you disable a checked radio button?